### DISCUSSION TITLE
How to achieve client-side data fetching on client-side routing

### DISCUSSION THREAD

**Discussion by debjits1** [vercel/next.js#53093] [Help]
### Summary

I want server-side rendering and server-side data fetching only on the first page load. All of the client-side navigation should use client-side data fethcing. I am currently using `getInitialProps` for the same. But I want to completely move to app directories and don not want to use `getInitialProps` since it is deprecated. How do I achieve the same behavior in next.js v13?

**Comment by edsphinx** [comment]
have the same question
